processors/base/helpers.py

```diff
@@ -202,9 +202,12 @@ def find_trial_by_identifiers(conn, identifiers):
     for source, identifier in identifiers.items():
         query = QUERY % json.dumps({source: identifier})
         records = list(conn['database'].query(query))
-        if not records:
-            continue
-        trial = conn['database']['trials'].find_one(id=records[0]['trial_id'].hex)
+        for record in records:
+            if not record['trial_id']:
+                continue
+            trial = conn['database']['trials'].find_one(id=record['trial_id'].hex)
+            if trial:
+                break
```

Comment:
The query could be simplified to:

`SELECT records.trial_id, identifier.key, identifier.value FROM records, json_each_text(records.identifiers::json) AS identifier WHERE identifier.value IN ('foo', 'bar')`

This doesn't take in consideration the identifiers' source. To do so, we could change the query to something like:

`SELECT records.trial_id FROM records WHERE trial_id IS NOT NULL AND (identifiers @> '{"nct": "NCT00322725"}' OR identifiers @> '{"nct": "FOO"}')`

```python
base_query = 'SELECT trial_id FROM records WHERE trial_id IS NOT NULL AND ({where_clause})'
where_clause = ' OR '.join(['identifiers @> ?'] * len(identifiers.keys()))
query = base_query.format(where_clause=where_clause)

identifiers_clause = [json.dumps({source: identifier})
                      for source, identifier in identifiers.items()]

records = conn['database'].query(query, identifiers_clause)
if records:
    # No need to test if the "trial_id" exists, as we're protected by
    # Postgres' foreign key constraints
    return conn['database']['trials'].find_one(id=records[0]['trial_id'].hex)
```

This reduces the number of queries we need to do to 2, regardless on how many identifiers a trial has. We could even get the trial in a single query, but that might be overkill.